### DISCUSSION TITLE
部分操作方式加入稍后读时，无法触发自动化

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2362] [重要通知]
## 描述

简悦2.2增加了自动化功能，可以实现将文章`加入稍后读`/`标注`后，触发自动化操作来实现自定义的导出操作等。

## 问题

目前 `加入稍后读` 触发自动化存在bug，使用 `加入稍后读`  功能时，部分操作方式无法触发自动化。

操作是否可以触发自动化：

- [x] 在阅读模式下按快捷键 `D D` （ 快捷键可更改）
- [x] 右键菜单 → 将当前页面加入稍后读
- [x] [使用 Assistive Touch 插件实现加入稍后读自动化](https://github.com/Kenshin/simpread/discussions/2362#discussioncomment-4233529)
- [ ] 在阅读模式下按快捷键 `SR` 启用快捷键模式，再按 `RL` 
- [ ] 在阅读模式下，在 `简悦选项卡 - 动作` 中，选择 `简悦 · 稍后读`

> 在下一版本修复前，请使用快捷键 `D D` 或 **右键菜单** 执行 `加入稍后读` 功能

## 使用自动化的前提

实现自动化的前提是：**进入阅读模式后**。**未进入阅读模式时，仅可以加入稍后读，无法触发自动化**。

因为：使用自动化的前提需要获取（阅读模式的）正文，在未进入阅读模式前，无法正确获取正文，所以无法使用自动化。

## 备注

因聚焦模式无法使用稍后读 / 标注，所以此功能不适合聚焦模式。

以上问题 **仅存在于使用触发稍后读后的自动化** 方案，其余并无影响。

## 关联

下个版本会解决上述 Bug，关联 Issues https://github.com/Kenshin/simpread/issues/2316

## 一个新场景

这个 Bug 可以引申为另外一个功能：**即有些 URL 只想加入稍后读并不需要触发自动化的场景**，比如：只是将某些 URL 做暂存用。

**Comment by handbye** [comment]
今天测试使用tg bot保存到稍后读后也无法触发自动化，还望修复这个bug，急需使用，谢谢。 @Kenshin

**Reply by Kenshin** [reply]
嗨，通过 telegram bot 保存到稍后读的文章并不会触发自动化。

自动化只是扩展端行为。

**Comment by invisprints** [comment]
补充下，在进入阅读模式后，首次使用标注无法触发加入到稍后读的自动化

**Reply by Kenshin** [reply]
哈哈，我认识你的 ID 了~

PS：查了下，哈哈果然是老用户 19年就提过 Issues~

***

> 首次使用标注无法触发加入到稍后读的自动化

应该没有这个问题，方便描述下加入过程吗？比如使用快捷键还是什么其它方式等？

**Reply by invisprints** [reply]
比如在自动化中只设置“当加入 `稍后读` 后导出 `HTML` ”，然后在简悦的阅读模式下直接开始批注，这时候简悦会把文章加入到稍后读里，但是并没有同时执行这个导出 `HTML` 自动化操作

**Reply by Kenshin** [reply]
> 比如在自动化中只设置“当加入 稍后读 后导出 HTML ”，然后在简悦的阅读模式下直接开始批注，这时候简悦会把文章加入到稍后读里，但是并没有同时执行这个导出 HTML 自动化操作

哈哈，这是正常的，你需要加上标注的自动化而不仅仅只是稍后读的自动化

![image](https://user-images.githubusercontent.com/81074/157157772-56a3d9cc-96d1-4355-96a4-9abcc51faac5.png)

**Comment by Kenshin** [comment]
# 使用  Assistive Touch 插件实现加入稍后读自动化

可以使用 [Assistive Touch](https://github.com/Kenshin/simpread/discussions/1600) 来更方便的加入稍后读并触发自动化，只需要按下图所示设置即可。

<img width="856" alt="image" src="https://user-images.githubusercontent.com/81074/196600141-24aa911a-5d75-4596-a4ad-c01928c8d440.png">
